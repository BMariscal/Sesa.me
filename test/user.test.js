import React from 'react';
import Login from '../client/src/components/user/formView/login.jsx';
import Signup from '../client/src/components/user/formView/signup.jsx';
import Response from '../client/src/components/user/formView/response.jsx';
import Submission from '../client/src/components/user/formView/submission.jsx';
import UserResponses from '../client/src/components/user/formView/userResponses.jsx';
import renderer from 'react-test-renderer';

/*TEST USER COMPONENTS*/

test('Login component renders correctly', () => {
  const tree = renderer
  .create(<Login></Login>)
  .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Signup component renders correctly', () => {
  const tree = renderer
  .create(<Signup></Signup>)
  .toJSON();
  expect(tree).toMatchSnapshot();
});

//TODO: response.jsx

//TODO: submission.jsx

//TODO: userResponse.jsx

