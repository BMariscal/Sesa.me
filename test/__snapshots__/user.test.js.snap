// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders correctly 1`] = `
<div
  className="container login-container"
>
  <div>
    <div
      className="page-header"
    >
      <h1>
        <small>
          Login to report a bug:
        </small>
      </h1>
    </div>
    <label
      className="login-username control-label"
      htmlFor={undefined}
    >
      Username
      <input
        className="form-control"
        id={undefined}
        onChange={[Function]}
        placeholder="username..."
        type="text"
      />
    </label>
    <br />
    <label
      className="login-password control-label"
      htmlFor={undefined}
    >
      Password
      <input
        className="form-control"
        id={undefined}
        onChange={[Function]}
        placeholder="password..."
        type="password"
      />
    </label>
    <br />
    <button
      className="login-button btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Log In
    </button>
    <br />
  </div>
  <div>
    <div
      className="page-header"
    >
      <h1>
        <small>
          Don't have an account? Sign up:
        </small>
      </h1>
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={undefined}
      type="button"
    >
      Signup
    </button>
  </div>
</div>
`;

exports[`Signup component renders correctly 1`] = `
<div
  className="container signup-container"
>
  <div
    className="page-header"
  >
    <h1>
      <small>
        Signup
      </small>
    </h1>
  </div>
  <label
    className="signup-username control-label"
    htmlFor={undefined}
  >
    Username
    <input
      className="form-control"
      id={undefined}
      onChange={[Function]}
      placeholder="username..."
      type="text"
    />
  </label>
  <br />
  <label
    className="signup-username control-label"
    htmlFor={undefined}
  >
    First Name
    <input
      className="form-control"
      id={undefined}
      onChange={[Function]}
      placeholder="first name..."
      type="text"
    />
  </label>
  <br />
  <label
    className="signup-username control-label"
    htmlFor={undefined}
  >
    Last Name
    <input
      className="form-control"
      id={undefined}
      onChange={[Function]}
      placeholder="last name..."
      type="text"
    />
  </label>
  <br />
  <label
    className="signup-password control-label"
    htmlFor={undefined}
  >
    Password
    <input
      className="form-control"
      id={undefined}
      onChange={[Function]}
      placeholder="password..."
      type="password"
    />
  </label>
  <br />
  <label
    className="signup-password control-label"
    htmlFor={undefined}
  >
    Type Password Again
    <input
      className="form-control"
      id={undefined}
      onChange={[Function]}
      placeholder="password..."
      type="password"
    />
  </label>
  <br />
  <div
    className="col-centered"
  >
    <div
      className="btn-toolbar"
      role="toolbar"
    >
      <button
        className="sign-up-button btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Create Account
      </button>
      <button
        className="sign-up-button btn btn-primary"
        disabled={false}
        onClick={undefined}
        type="button"
      >
        Return to log in page
      </button>
    </div>
  </div>
</div>
`;

exports[`Submission component renders correctly 1`] = `
<div
  className="user-submission-main"
>
  <div
    className="page-header"
  >
    <h1>
      Submission Form
    </h1>
  </div>
  <h5>
    Use this form to submit a bug or to respond to an admin's request for additional information.
  </h5>
  <button
    className="change-view-button btn btn-primary"
    disabled={false}
    onClick={undefined}
    type="button"
  >
    View Correspondence
  </button>
  <div
    className="user-submission-container"
  >
    <label
      className="contact-info control-label"
      htmlFor={undefined}
    >
      How can we contact you?
    </label>
    <br />
    <textarea
      className="form-control"
      id={undefined}
      onChange={[Function]}
      placeholder="contact information"
      type="text"
    />
    <br />
    <div>
      <label
        className="control-label"
        htmlFor={undefined}
      >
        Please rank the urgency of this bug on a scale of 1 through 5, where 1 is the least urgent, and 5 indicates that the bug is of immediate urgency.
      </label>
      <br />
      <span>
        1
      </span>
      <input
        name="urgency"
        onClick={[Function]}
        type="radio"
        value="1"
      />
      <span>
        2
      </span>
      <input
        name="urgency"
        onClick={[Function]}
        type="radio"
        value="2"
      />
      <span>
        3
      </span>
      <input
        name="urgency"
        onClick={[Function]}
        type="radio"
        value="3"
      />
      <span>
        4
      </span>
      <input
        name="urgency"
        onClick={[Function]}
        type="radio"
        value="4"
      />
      <span>
        5
      </span>
      <input
        name="urgency"
        onClick={[Function]}
        type="radio"
        value="5"
      />
    </div>
    <br />
    <label
      className="message control-label"
      htmlFor={undefined}
    >
      Please provide any information about the bug that we may use to help you:
    </label>
    <br />
    <textarea
      className="form-control"
      id={undefined}
      onChange={[Function]}
      placeholder="Additional information..."
      type="text"
    />
    <br />
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Submit Message
    </button>
  </div>
</div>
`;
